src/RemoteAPI/class-analytics-post-detail-api.php

```diff
@@ -18,7 +18,7 @@
  * @since 3.6.0
  */
 class Analytics_Post_Detail_API extends Remote_API_Base {
-	protected const ENDPOINT     = Parsely::PUBLIC_API_BASE_URL . '/analytics/post/detail';
```

Comment:
- Removed this base URL from here so that we can change the endpoint to filter key in more cleaner way.
- Added this base URL inside `get_api_url` function of Remote_Base_API (only needed once because its a common location)